.github/workflows/ci_workflow.yml

```diff
@@ -0,0 +1,397 @@
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+        with:
+          fetch-depth: 50
+      - name: Create build directories
+        run: |
+          mkdir _install
+          mkdir _build
+      - name: Configure
+        run: |
+          cmake ../. \
+                -DCMAKE_INSTALL_PREFIX=../_install \
+                -DCMAKE_BUILD_TYPE=Release \
+                -DCMAKE_CXX_STANDARD=14 \
+                -DCMAKE_CXX_FLAGS="-g -O0 -fprofile-arcs -ftest-coverage" \
+                -DCMAKE_CXX_OUTPUT_EXTENSION_REPLACE=ON \
+                -DCMAKE_EXE_LINKER_FLAGS="-lgcov" \
+                -DOCIO_BUILD_DOCS=ON \
+                -DOCIO_BUILD_GPU_TESTS=OFF \
+                -DOCIO_INSTALL_EXT_PACKAGES=ALL \
+                -DOCIO_WARNING_AS_ERROR=ON
+        working-directory: _build
+      - name: Build OCIO with build-wrapperr
+        run: build-wrapper-linux-x86-64 --out-dir bw_output make clean all
+        working-directory: _build
+      - name: Test
+        run: ctest -V
+        working-directory: _build
+      - name: Generate code coverage report
+        run: share/ci/scripts/linux/run_gcov.sh
+      - name: Run sonar-scanner
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          SONAR_TOKEN: ${{ secrets.SONAR_TOKEN }}
+        run: sonar-scanner -X -Dsonar.login=$SONAR_TOKEN
+    
+  # ---------------------------------------------------------------------------
+  # macOS
+  # ---------------------------------------------------------------------------
+
+  macos:
+    name: macOS 10.15 AppleClang 11.0 ${{ matrix.build-desc }}
+    runs-on: macos-10.15
+    strategy:
+      matrix:
+        build: [1, 2, 3]
+        include:
+          - build: 1
+            build-desc: ''
+            build-type: Release
+            python-version: 3.7
+            shared: 'ON'
+            use-sse: 'ON'
+          - build: 2
+            build-desc: (Debug)
+            build-type: Debug
+            python-version: 3.7
+            shared: 'ON'
+            use-sse: 'ON'
+          - build: 3
+            build-desc: (static, no SSE, Python 2.7)
+            build-type: Release
+            python-version: 2.7
+            shared: 'OFF'
+            use-sse: 'OFF'
+    steps:
+      - name: Setup Python
+        uses: actions/setup-python@v1
+        with:
+          python-version: ${{ matrix.python-version }}
+      - name: Checkout
+        uses: actions/checkout@v2
+      - name: Install dependencies
+        run: share/ci/scripts/macos/install.sh
+      - name: Create build directories
+        run: |
+          mkdir _install
+          mkdir _build
+      - name: Configure
+        run: |
+          cmake ../. \
+                -DCMAKE_INSTALL_PREFIX=../_install \
+                -DCMAKE_BUILD_TYPE=${{ matrix.build-type }} \
+                -DCMAKE_CXX_STANDARD=14 \
+                -DBUILD_SHARED_LIBS=${{ matrix.shared }} \
+                -DOCIO_BUILD_DOCS=ON \
```

Comment:
Useful to have at least one build generating the documentation, but not needed for all builds.